.github/workflows/scripts/npm_release_files/ts_serialize.d.ts

```diff
@@ -1,97 +1,89 @@
     public tsTransformKey?(key: string): string;
-    /** Serializable to Json String */
-    public toJson(): string;
-    /** Deserialize to Serializable */
-    public fromJson(json: JsonValue | Object): this;
+    /** to JSON String */
+    public toJSON(): string;
+    /** to Serializable */
+    public fromJSON(json: JSONValue | Object): this;
+    /** to JSONObject */
+    public tsSerialize(): JSONObject;
   }
 
   /** Functions used when hydrating data */
-  export type FromJsonStrategy = (value: JsonValue) => any;
-  export type FromJsonStrategyArgument =
-    (FromJsonStrategy | FromJsonStrategy[])[];
+  export type FromJSONStrategy = (value: JSONValue) => any;
+  export type FromJSONStrategyArgument =
+    (FromJSONStrategy | FromJSONStrategy[])[];
 
   /** Functions used when dehydrating data */
-  export type ToJsonStrategy = (value: any) => JsonValue;
-  export type ToJsonStrategyArgument = (ToJsonStrategy | ToJsonStrategy[])[];
+  export type ToJSONStrategy = (value: any) => JSONValue;
+  export type ToJSONStrategyArgument = (ToJSONStrategy | ToJSONStrategy[])[];
 
   /** string/symbol property name or options for (de)serializing values */
   export type SerializePropertyArgument =
     | string
     | {
       serializedKey?: string;
-      fromJsonStrategy?:
-        | FromJsonStrategy
-        | FromJsonStrategyArgument;
-      toJsonStrategy?:
-        | ToJsonStrategy
-        | ToJsonStrategyArgument;
+      fromJSONStrategy?:
+        | FromJSONStrategy
+        | FromJSONStrategyArgument;
+      toJSONStrategy?:
+        | ToJSONStrategy
+        | ToJSONStrategyArgument;
     };
 
-  /** Property wrapper that adds serializable options to the class map
-   * using the original propertyName as the map key
-   */
-  export function SerializeProperty(): PropertyDecorator;
-
-  /** Property wrapper that adds serializable options to the class map
-   * using the provided string as the map key
-   */
-  export function SerializeProperty(arg: string): PropertyDecorator;
-
   /** Property wrapper that adds serializable options to the class map
    * using options, `serializedName` as the key or `propertyName` if
    * `serializedName` is not set
    */
   export function SerializeProperty(
-    arg: SerializePropertyArgument,
+    arg: string | SerializePropertyArgument,
   ): PropertyDecorator;
 
-  /** Function to build a `fromJsonStrategy` or `toJsonStrategy`.
+  /** Function to build a `fromJSONStrategy` or `toJSONStrategy`.
    * Converts value from functions provided as parameters
    */
   export function composeStrategy(
     ...fns:
-      | FromJsonStrategyArgument
-      | ToJsonStrategyArgument
-  ): FromJsonStrategy | ToJsonStrategy;
+      | FromJSONStrategyArgument
+      | ToJSONStrategyArgument
+  ): FromJSONStrategy | ToJSONStrategy;
 
-  /** revive data using `fromJson` on a subclass type */
-  export function fromJsonAs<T>(
+  /** revive data using `fromJSON` on a subclass type */
+  export function as<T>(
```

Comment:
I worry a bit about polluting the caller's namespace with a generic name like `as()`, it seems kind of like rxjs's operator names.

Reply:
> it seems kind of like rxjs's

ya it is silly because I was going to comment with "oh they can just change via the `as` keyword"  so now I am thinking